server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -1607,6 +1608,28 @@ public Builder generateClusterUuidIfNeeded() {
             return this;
         }
 
+        private void canonicalizeMappingsInPlace() {
```

Comment:
I don't see a case where we are deleting a index, because it can happen that the mappings object we are holding in memory is deleted as part of the delete index call, then all other indices with same mapping pointing to null effectively.

Is that not the case ?

Reply:
The logic added here is within the metadata builder so the indices object in this builder would be only the present indices. I need to check, but on index deletion it would create a new version of the metadata using this builder with all of the indices except the one that has been deleted. This function is then called in `build()` to de-dup the mappings so that we don't hold multiple versions of the same mappings in-memory for the new in-memory Metadata.

Reply:
Relevant code: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/metadata/MetadataDeleteIndexService.java#L159-L169